### PR TITLE
Support python 3

### PR DESCRIPTION
**Pull request by matburt**
* Correct use of threading
* Fix up encoding issue on event partials
* Fix up dictionary modification when processing passwords and event data

### CODE REVIEW

File: ansible_runner/utils.py [hunk 1]
```diff
@@ -9,7 +8,7 @@
 import hashlib
 
 from collections import Iterable, Mapping
-
+from io import StringIO
```
Comment (cchurch) [inline ansible_runner/utils.py]: `six.StringIO` unless you're planning on dropping Python 2.7 support entirely.

File: ansible_runner/runner_config.py [hunk 2]
```diff
@@ -177,8 +177,8 @@ def open_fifo_write(self, path, data):
         reads data from the pipe.
         '''
         os.mkfifo(path, 0o600)
-        thread.start_new_thread(lambda p, d: open(p, 'w').write(d), (path, data))
-
+        threading.Thread(target=lambda p, d: open(p, 'w').write(d),
+                         args=(path, data))
```
Comment (cchurch) [inline ansible_runner/runner_config.py]: Also need to call `start()` on the resulting `Thread` instance.

File: ansible_runner/utils.py (hunk 1 above)
Comment (matburt) [reply ansible_runner/utils.py]: It looks like `io.StringIO` exists in 2.7 at least?

File: ansible_runner/runner_config.py (hunk 2 above)
Comment (matburt) [reply ansible_runner/runner_config.py]: good catch

**Pull request merged**